### PR TITLE
bpo-39019: Implement missing __class_getitem__ for subprocess classes

### PR DESCRIPTION
**Pull request by isidentical**
(It also includes a fix for a news entry from a previous PR, GH-17498)

https://bugs.python.org/issue39019

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2019-12-07-18-58-44.bpo-38994.IJYhz_.rst [hunk 1]
```diff
@@ -1 +1 @@
-Implement ``__class_getitem__`` for ``os.PathLike``, ``pathlib.Path``
+Implement ``__class_getitem__`` for ``os.PathLike``, ``pathlib.PurePath``. Patch by Batuhan Taskaya.
```
Comment (ilevkivskyi) [inline Misc/NEWS.d/next/Library/2019-12-07-18-58-44.bpo-38994.IJYhz_.rst]: This change looks unrelated.

File: Misc/NEWS.d/next/Library/2019-12-10-21-03-34.bpo-39019.i8RpMZ.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Implement ``__class_getitem__`` for ``subprocess.Popen``,
```
Comment (ilevkivskyi) [inline Misc/NEWS.d/next/Library/2019-12-10-21-03-34.bpo-39019.i8RpMZ.rst]:
```suggestion
Implement dummy ``__class_getitem__`` for ``subprocess.Popen``,
```

File: Lib/subprocess.py [hunk 3]
```diff
@@ -446,6 +446,9 @@ def __repr__(self):
             args.append('stderr={!r}'.format(self.stderr))
         return "{}({})".format(type(self).__name__, ', '.join(args))
 
+    def __class_getitem__(cls, type):
+        return cls
```
Comment (ilevkivskyi) [inline Lib/subprocess.py:459]: Here and below I would add a short docstring explaining why this is needed, and why it is implemented in such trivial way.

**Review by ilevkivskyi — COMMENTED** [review]
Thanks! Here are few comments.

File: Misc/NEWS.d/next/Library/2019-12-07-18-58-44.bpo-38994.IJYhz_.rst (hunk 1 above)
Comment (isidentical) [reply Misc/NEWS.d/next/Library/2019-12-07-18-58-44.bpo-38994.IJYhz_.rst]: It was a mistake from a previous patch and I didn't want to open a new PR for that. I explained it in PR body.

Comment (ilevkivskyi) [reply Misc/NEWS.d/next/Library/2019-12-07-18-58-44.bpo-38994.IJYhz_.rst]: OK, anyway, it looks like there is a merge conflict now, you need to fix it before we can move on.

Comment (isidentical) [reply Misc/NEWS.d/next/Library/2019-12-07-18-58-44.bpo-38994.IJYhz_.rst]: Thanks for the reviews. I'm applying them.

File: Lib/subprocess.py (hunk 3 above)
Comment (isidentical) [reply Lib/subprocess.py:459]: What kind of docstring we should put? `"""For using this class as a generic, it implements PEP 560 __class_getitem__ method"""`?

Comment (ilevkivskyi) [reply Lib/subprocess.py:459]: I think this should be something a bit more detailed. For example:
```python
def __class_getitem__(cls, type):
    """Provide minimal support for using this class as generic (for example in type annotations).

    See PEP 484 and PEP 560 for more details. For example, `CompletedProcess[bytes]` is a valid
    expression at runtime (type argument `bytes` indicates the type used for stdout). Note, no
    type checking happens at runtime, but a static type checker can be used.
    """
    return cls
```

Comment (isidentical) [reply Lib/subprocess.py:459]: I see thanks.

**Review by ilevkivskyi — APPROVED** [review]
LGTM!

**Comment by ilevkivskyi** [comment]
@gpshead I am going to merge this soon unless you are against it.

**Review by gpshead — APPROVED** [review]

**Comment by ilevkivskyi** [comment]
Unless I am missing something, the buildbot failure looks like a flake.
